thirdparty/protobuf/CMakeLists.txt

```diff
@@ -58,32 +57,22 @@ target_include_directories(libprotobuf-lite SYSTEM PRIVATE "${Protobuf_INCLUDE_D
 if(CMAKE_COMPILER_IS_GNUCXX OR OV_COMPILER_IS_CLANG OR
    CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     if(protobuf_BUILD_PROTOC_BINARIES)
-        set_target_properties(protoc libprotoc PROPERTIES
+        list(APPEND _protoc_libs protoc libprotoc libprotobuf)
+        set_target_properties(${_protoc_libs} PROPERTIES
             CXX_VISIBILITY_PRESET default
             C_VISIBILITY_PRESET default
-            VISIBILITY_INLINES_HIDDEN OFF)
-        list(APPEND _protobuf_libs protoc libprotoc)
+            VISIBILITY_INLINES_HIDDEN OFF
+            INTERPROCEDURAL_OPTIMIZATION_RELEASE OFF)
     endif()
-    ov_disable_all_warnings(${_protobuf_libs} libprotobuf libprotobuf-lite)
-endif()
-
-if(NGRAPH_USE_PROTOBUF_LITE)
-    # if only libprotobuf-lite is used, both libprotobuf and libprotobuf-lite are built
-    # libprotoc target needs symbols from libprotobuf, even in libprotobuf-lite configuration
-    set_target_properties(libprotobuf PROPERTIES
-        CXX_VISIBILITY_PRESET default
-        C_VISIBILITY_PRESET default
-        VISIBILITY_INLINES_HIDDEN OFF)
+    ov_disable_all_warnings(${_protoc_libs} libprotobuf-lite)
+    set_target_properties(libprotobuf-lite PROPERTIES
+        INTERPROCEDURAL_OPTIMIZATION_RELEASE ${ENABLE_LTO})
```

Comment:
LTO is always disabled for protobuf libs but protobuf-lite have it enabled conditionally.
Could you explain why we have to do it this way?
What if someone will use a full protobuf to build OV, should it be enabled for it as well?

Reply:
> LTO is always disabled for protobuf libs but protobuf-lite have it enabled conditionally.
Could you explain why we have to do it this way?

We had code like this:
```cmake
if(ENABLE_LTO AND protobuf_VERSION VERSION_GREATER_EQUAL "3.8")
    message(WARNING "Protobuf in version 3.8.0+ can throw runtime exceptions if LTO is enabled.")
endif()
```
So, we have exception if we compile `protobuf`, so we always compile `protobuf` without LTO.
But don't have such issues with `protobuf-lite` so we can compile it with LTO if it's enabled.

> What if someone will use a full protobuf to build OV, should it be enabled for it as well?
Is it possible to do that? I don't see that we can find only protobuf, but no protobuf lite https://cmake.org/cmake/help/latest/module/FindProtobuf.html

Reply:
by **it** I meant LTO.
Everything is clear, thanks.